nemo/collections/asr/data/audio_to_text_dataset.py

```diff
@@ -299,6 +307,47 @@ def get_dali_bpe_dataset(
     return dataset
 
 
+class ASRPredictionWriter(BasePredictionWriter):
+    def __init__(self, dataset, output_file: str, use_cer=False):
+        super().__init__(write_interval="batch")
+        self.outf = open(output_file, 'w')
+        self.dataset = dataset
+        self.use_cer = use_cer
+        self.samples_num = 0
+
+    def write_on_batch_end(
+        self,
+        trainer,
+        pl_module: 'LightningModule',
+        prediction: Any,
+        batch_indices: List[int],
+        batch: Any,
+        batch_idx: int,
+        dataloader_idx: int,
+    ):
+        for sample_id, transcribed_text in prediction:
+            item = {}
+            sample = self.dataset.get_sample(sample_id)
+            item["audio_filepath"] = sample.audio_file
+            item["duration"] = sample.duration
+            item["text"] = sample.text_raw
+            item["pred_text"] = transcribed_text
+            wer_cer = word_error_rate(
```

Comment:
This is both correct in one sense and wrong in another- this will cause "average of averages" behaviour when calculating metrics if you take the average of all the sample WERs across number of samples in the dataset. 

It is fine to calculate just the single sample level WER/CER though. 
If you need to use it to calculate the "global dataset WER/CER" then you need to create the appropriate CTC/RNNT WER/WERBPE metric object, call it on the predictions of the model, and globally preserve the scores and words tensor (sum across batches) and then do a final division at the end. Make a note of this or remove these two metrics entirely from the logs. It causes more confusion than solving the problem and given a text manifest output it is easy to calculate sample or global metrics anyway.

Reply:
Currently we do not log any wer, this one just calculates the wer per sample and it would be written into the output file. We can add a global wer calculation but let's add it in another PR.

Reply:
WER per sample is "wrong" though. People will try to calculate WER global by averaging WER per sample and get wrong results. Please revert or compute global in this PR only.